app/models/process_pending_portal_publication.rb

```diff
@@ -0,0 +1,40 @@
+class ProcessPendingPortalPublication < Struct.new(:pending_portal_publication_id, :auto_publish_url, :backoff)
+  def perform
+    # immediately find and remove the pending publication so that new publications can be queued
+    pending_portal_publication = PendingPortalPublication.find(pending_portal_publication_id)
+    return if pending_portal_publication.nil?
+    pending_portal_publication.delete
+
+    # get shorter local references
+    portal_publication = pending_portal_publication.portal_publication
+    return if portal_publication.nil?
+    publishable = portal_publication.publishable
+    return if publishable.nil?
+    portal = Concord::AuthPortal.portal_for_publishing_url(portal_publication.portal_url)
+    last_portal_publication = publishable.last_publication(portal)
+    return if last_portal_publication.nil?
+
+    # if the portal publication is not the latest one another job has published behind us so we are done
+    return if portal_publication.id != last_portal_publication.id
+
```

Comment:
I think this should find the last successful publication.  There might be a failed publication that occurs in between.